account_invoice_import_tensorflow/models/account_invoice_import.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2020 Hunki Enterprises BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import json
+import requests
+from odoo import models
+from odoo.tools import float_round
+
+
+class AccountInvoiceImport(models.TransientModel):
+    _inherit = "account.invoice.import"
+
+    def fallback_parse_pdf_invoice(self, file_data):
+        parsed_data = self._account_invoice_import_tensorflow_parse(file_data)
+        if parsed_data.get('failed'):
+            return super(AccountInvoiceImport, self).fallback_parse_pdf_invoice(file_data)
+        return parsed_data
+
+    def _account_invoice_import_tensorflow_parse(self, file_data):
+        """Send file to serving container and return its data"""
+        #response = requests.post(
+        #    self.env['ir.config_parameter'].get_param(
+        #        'account_invoice_import_tensorflow.predict_url'
+        #    ),
+        #    json=self._account_invoice_import_tensorflow_predict_data(
+        #        file_data
+        #    ),
+        #)
+        # return self._account_invoice_import_tensorflow_parse_response(response.json())
+        response_data = dict(predictions=json.loads(file_data))
+        if not response_data['predictions']:
+            # TODO how does a response of a failed prediction look?
+            return dict(failed=True)
+        return self._account_invoice_import_tensorflow_parse_response(response_data)
+
+    def _account_invoice_import_tensorflow_predict_data(self, file_data):
+        """Return data passed to the serving predict endpoint"""
+        # TODO how do we need to format our input?
+        return {
+            "inputs": base64.b64encode(file_data),
+        }
+
+    def _account_invoice_import_tensorflow_parse_response(self, response):
+        """Return data usable as result of account.invoice.import#parse_pdf_invoice"""
+        data = response['predictions']
+        # TODO will the model ever recogize the currency?
```

Comment:
@Stephanvandenberg are currencies always one-to-one linked to vendors? If so, we could make this deterministic, otherwise we could put this on our backlog and let the model predict this (in a later phase)